4-projects/business_unit_2/dev/example_single_project_optional.tf

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# module "example_single_project_optional" {
+#   source = "../../modules/single_project"
+
+#   org_id                      = var.org_id
+#   billing_account             = var.billing_account
+#   impersonate_service_account = var.terraform_service_account
+#   folder_id = var.parent_folder
+#   environment = "dev"
+
+#   # Metadata
+#   project_prefix   = "single-optional-bu2-d"
+#   cost_centre      = "cost-centre-1"
+#   application_name = "sample-single-project-app-optional-bu2-d"
+
+#   # Network Setting (Optional)
+#   enable_networking    = true
```

Comment:
I think we should have some of these optionals ( and optional modules) enabled in some env so that we can test these code paths as well. @rjerrems WDYT?

Reply:
Yeah this is a good idea, although the DNS module will need some updates with the new DNS hub architecture, but we can deal with that separately.